pkg/ingester/wal.go

```diff
@@ -29,21 +29,21 @@ import (
 
 // WALConfig is config for the Write Ahead Log.
 type WALConfig struct {
-	walEnabled         bool          `yaml:"wal_enabled,omitempty"`
-	checkpointEnabled  bool          `yaml:"checkpoint_enabled,omitempty"`
-	recover            bool          `yaml:"recover_from_wal,omitempty"`
-	dir                string        `yaml:"wal_dir,omitempty"`
-	checkpointDuration time.Duration `yaml:"checkpoint_duration,omitempty"`
+	WalEnabled         bool          `yaml:"wal_enabled,omitempty"`
+	CheckpointEnabled  bool          `yaml:"checkpoint_enabled,omitempty"`
+	Recover            bool          `yaml:"recover_from_wal,omitempty"`
+	Dir                string        `yaml:"wal_dir,omitempty"`
+	CheckpointDuration time.Duration `yaml:"checkpoint_duration,omitempty"`
 	metricsRegisterer  prometheus.Registerer
```

Comment:
Despite correct, please also add `yaml:"-"` to enforce we don't want it to be configurable.